apps/stripe/src/modules/transactions-recording/repositories/dynamodb/dynamodb-transaction-recorder-repo.ts

```diff
@@ -0,0 +1,152 @@
+import { GetItemCommand, PutItemCommand } from "dynamodb-toolbox";
+import { err, ok, Result } from "neverthrow";
+
+import { BaseError } from "@/lib/errors";
+import { createLogger } from "@/lib/logger";
+import { createResolvedTransactionFlow } from "@/modules/resolved-transaction-flow";
+import { createSaleorTransactionFlow } from "@/modules/saleor/saleor-transaction-flow";
+import { createSaleorTransactionId } from "@/modules/saleor/saleor-transaction-id";
+import { PaymentMethod } from "@/modules/stripe/payment-methods/types";
+import {
+  createStripePaymentIntentId,
+  StripePaymentIntentId,
+} from "@/modules/stripe/stripe-payment-intent-id";
+import { RecordedTransaction } from "@/modules/transactions-recording/domain/recorded-transaction";
+import {
+  DynamoDbRecordedTransaction,
+  DynamoDbRecordedTransactionEntity,
+} from "@/modules/transactions-recording/repositories/dynamodb/recorded-transaction-db-model";
+import {
+  TransactionRecorderError,
+  TransactionRecorderRepo,
+  TransactionRecorderRepoAccess,
+} from "@/modules/transactions-recording/repositories/transaction-recorder-repo";
+
+export class DynamoDBTransactionRecorderRepo implements TransactionRecorderRepo {
+  private entity: DynamoDbRecordedTransactionEntity;
+
+  private logger = createLogger("DynamoDBTransactionRecorderRepo");
+
+  constructor(
+    params = {
+      entity: DynamoDbRecordedTransaction.entity,
+    },
+  ) {
+    this.entity = params.entity;
+  }
+
+  // TODO: Check against duplicate, do not allow to write it
+  async recordTransaction(
+    accessPattern: TransactionRecorderRepoAccess,
+    transaction: RecordedTransaction,
+  ): Promise<Result<null, TransactionRecorderError>> {
+    try {
+      this.logger.debug("Trying to write Transaction to DynamoDB", { transaction });
+
+      const result = await this.entity
+        .build(PutItemCommand)
+        .item({
+          PK: DynamoDbRecordedTransaction.accessPattern.getPK({
+            appId: accessPattern.appId,
+            saleorApiUrl: accessPattern.saleorApiUrl,
+          }),
+          SK: DynamoDbRecordedTransaction.accessPattern.getSKforSpecificItem({
+            paymentIntentId: transaction.stripePaymentIntentId,
+          }),
+          paymentIntentId: transaction.stripePaymentIntentId,
+          selectedPaymentMethod: transaction.selectedPaymentMethod,
+          saleorTransactionId: transaction.saleorTransactionId,
+          saleorTransactionFlow: transaction.saleorTransactionFlow,
+          resolvedTransactionFlow: transaction.resolvedTransactionFlow,
+        })
```

Comment:
suggestion: regarding the TODO, we can check if item already exists by using [dynamodb toolbox options](https://www.dynamodbtoolbox.com/docs/entities/actions/put-item#examples):

```suggestion
        })
        .options({
          condition: { attr: 'paymentIntendId', exists: false },
        })
```